internal/parser_test.go

```diff
@@ -1076,16 +1077,16 @@ func Test_VariableAccess_ArraySyntax_DoesNotExist(t *testing.T) {
 
 func Test_VariableAccess_ArraySyntax_InvalidType(t *testing.T) {
 	vars := getTestVars()
-	assertEvalError(t, vars, "syntax error: object key must be string, but was bool", `obj[true]`)
-	assertEvalError(t, vars, "syntax error: object key must be string, but was number", `obj[0]`)
-	assertEvalError(t, vars, "syntax error: object key must be string, but was array", `obj[arr]`)
-	assertEvalError(t, vars, "syntax error: object key must be string, but was object", `obj[obj]`)
+	assertEvalError(t, vars, "type error: object key must be string, but was bool", `obj[true]`)
+	assertEvalError(t, vars, "type error: object key must be string, but was number", `obj[0]`)
+	assertEvalError(t, vars, "type error: object key must be string, but was array", `obj[arr]`)
+	assertEvalError(t, vars, "type error: object key must be string, but was object", `obj[obj]`)
 
-	assertEvalError(t, vars, "syntax error: array index must be number, but was bool", `arr[true]`)
```

Comment:
In my opinion this is more of a syntax error than a type error.

Reply:
The object key can also come from another variable:

`obj[var]`

This is syntactically correct. If `var` contains a number/float, the error is triggered. If it contains a string, everything is okay.
That's why I changed it to a type error.